indico/cli/i18n.py

```diff
@@ -113,62 +244,246 @@ def _wrapper(**kwargs):
+
+
+def _plugin_command(babel_cmd, python, javascript, react, locale, no_check, plugin_dir):
+    extra = {}
+    if locale:
+        extra['locale'] = locale
+    extra['plugin_dir'] = plugin_dir
+
+    with _chdir(plugin_dir):
+        click.secho(f'plugin: {plugin_dir}', fg='white', bold=True)
+        if not no_check:
+            translations_dir = _get_translations_dir(plugin_dir)
+            if not _check_format_strings(translations_dir):
+                click.secho(f'Exiting compile command for {plugin_dir} due to invalid format strings.', fg='red',
+                            bold=True, err=True)
+                return
+        try:
+            if python:
+                _run_command(babel_cmd, extra=extra)
+            if javascript:
+                _run_command(f'{babel_cmd}_js', extra=extra)
+            if react:
+                if babel_cmd == 'compile_catalog':
+                    compile_catalog_react(plugin_dir)
+                elif babel_cmd == 'extract_messages':
+                    extract_messages_react(plugin_dir)
+                else:
+                    _run_command(f'{babel_cmd}_react', extra=extra)
+        except Exception as err:
+            click.secho(f'Error running {babel_cmd} for {plugin_dir} - {err}', fg='red', bold=True, err=True)
+
+
+def _command(**kwargs):
+    babel_cmd = kwargs.get('babel_cmd')
+    python = kwargs.get('python')
+    javascript = kwargs.get('javascript')
+    react = kwargs.get('react')
+    locale = kwargs.get('locale')
+    plugin_dirs = kwargs.get('plugin_dirs')
+    no_check = kwargs.get('no_check', True)
+
+    if not (python or react or javascript):
+        python = react = javascript = True
+        if babel_cmd == 'compile_catalog':
+            javascript = False
+
+    if not plugin_dirs:
+        _indico_command(babel_cmd, python, javascript, react, locale, no_check)
+    else:
+        for plugin_dir in plugin_dirs:
+            _plugin_command(babel_cmd, python, javascript, react, locale, no_check, plugin_dir)
+
+
+def _make_command(group, cmd_name, babel_cmd, **kwargs):
+    cmd_group = group.command(cmd_name, short_help=f'Perform {babel_cmd} operation on {cmd_name}')
+
+    @_common_translation_options(**kwargs)
+    def wrapper(**kwargs):
+        _command(babel_cmd=babel_cmd, **kwargs)
+
+    return cmd_group(wrapper)
+
+
+_make_command(compile_catalog, 'indico', 'compile_catalog', require_js=False)
+_make_command(extract_messages, 'indico', 'extract_messages')
+_make_command(update_catalog, 'indico', 'update_catalog')
+_make_command(init_catalog, 'indico', 'init_catalog', require_locale=True)
+
+_make_command(compile_catalog, 'plugins', 'compile_catalog', require_js=False,
+              plugins=True)
+_make_command(extract_messages, 'plugins', 'extract_messages', plugins=True)
+_make_command(update_catalog, 'plugins', 'update_catalog', plugins=True)
+_make_command(init_catalog, 'plugins', 'init_catalog', require_locale=True,
+              plugins=True)
```

Comment:
This subcommand should be called `plugin` and not `plugins` since it's for a single plugin.

Reply:
Actually this command can take multiple plugin directories, that's why it's called plugins not plugin <dir1, dir1...>

Reply:
ah ok. still a but confusing considering that other scripts like build-assets.py use `plugin` (also, why would you want to build translations for some but not all plugins)

Reply:
Oh okay I'll change it, I just took the commands from the issue description

> I recommend having a look at the build-assets.py script and mostly mimic the behavior.
Instead of having separate CLI commands for core/plugins we could also do indico i18n compile indico [args], indico i18n compile all-plugins [args] <plugins-dir> and indico i18n compile plugins [args] <plugin-dirs...> (and likewise for other commands).

Reply:
I've changes this command to be called `plugin` and for it to accept only a single directory